tests/templates/test_integration.py

```diff
@@ -647,3 +647,82 @@ def circuit():
 
         # Check that execution does not throw error
         circuit()
+
+
+class TestNonConsecutiveWires:
+    """Tests that a template results in the same state whether we use nonconsecutive wire labels or not.
+    """
+
+    @pytest.mark.parametrize("template, diffable, nondiffable, n_wires", QUBIT_DIFFABLE_NONDIFFABLE)
+    def test_qubit_result_is_wire_label_independent(self, template, diffable, nondiffable, n_wires):
+        """Tests integration of qubit templates with non-integer and non-consecutive wires."""
+
+        # merge differentiable and non-differentiable arguments
+        nondiffable.update(diffable)
+
+        # construct qnode with consecutive wires
+        dev_consec = qml.device('default.qubit', wires=n_wires)
+
+        @qml.qnode(dev_consec)
+        def circuit_consec():
+            template(**nondiffable)
+            return qml.expval(qml.Identity(wires=0))
+
+        # construct qnode with nonconsecutive wires
```

Comment:
Good to fix the seed explicitly in the file, and verify that that particular seed gives nonconsecutive wires (small probability it wouldn't). If the seed is not set explicitly, there's a chance that the behaviour of some (stochastic) tests will be dependent on what other tests are run

Reply:
Good point actually! I decided to rather use a non-consecutive fixed base array from which the wires are sliced.

Reply:
Has this change been pushed? I still see an `np.random.choice` below?

Reply:
Sorry, not yet...refactoring this test is still a larger task!